src/batch.rs

```diff
@@ -30,8 +30,8 @@ impl Batch {
         self.0.len()
     }
 
-    pub fn sort(&mut self) {
-        self.0.sort_unstable_by(sort_by_meta);
+    pub fn sort(&mut self, sort: (SortFlag, SortOrder)) {
```

Comment:
You should pass the `Flags` struct directly here. Soon the variables inside `Flags` will be private and only accessible via getters in order to keep the flag priority logic into the flags package.